src/libexpr/flake/flake.cc

```diff
@@ -113,7 +113,7 @@ static FlakeInput parseFlakeInput(EvalState & state,
         try {
             if (attr.name == sUrl) {
                 expectType(state, nString, *attr.value, attr.pos);
-                url = attr.value->string.s;
+                url = attr.value->c_str();
```

Comment:
Shouldn't this use `->string_view()` instead of `->c_str()`, since it doesn't require a zero-terminated string? Generally only C APIs should require `c_str()`.

Reply:
Likely yes. My initial refactor just did a faithful string.s to c_str() conversion. But now it means we can change some of them to string_view().

Reply:
applied the suggestion